src/actions.ts

```diff
@@ -51,4 +55,44 @@ export async function registryLogin(): Promise<void> {
     }
 
     return spin.registryLogin(core.getInput('registry'), core.getInput('registry_username'), core.getInput('registry_password'))
+}
+
+export async function deployPreview(prNumber: number): Promise<cloud.Metadata> {
+    const manifestFile = getManifestFile()
+    const spinConfig = spin.getAppManifest(manifestFile)
+
+    const realAppName = spinConfig.name
+    const previewAppName = `${realAppName}-pr-${prNumber}`
+
+    core.info(`🚀 deploying preview as ${previewAppName}`)
+    const metadata = await cloud.deployAs(previewAppName, manifestFile)
+
+    const comment = `🚀 preview deployment successful and available at ${metadata.base}`
```

Comment:
Wondering if we want to add 'Spin' and/or 'Fermyon Cloud' in the comment?

Reply:
changed it to `preview deployed successfully to Fermyon Cloud and available at`

Reply:
I'm not seeing that change included in the latest, but we can always adjust/update later.

Reply:
i forgot to push the code :), pushed now.